bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/handler/HomeWizardP1MeterHandler.java

```diff
@@ -35,14 +38,17 @@ public class HomeWizardP1MeterHandler extends HomeWizardDeviceHandler {
 
     private String meterModel = "";
     private int meterVersion = 0;
+    private ZoneId zoneId;
 
     /**
      * Constructor
      *
      * @param thing The thing to handle
+     * @param zoneId The time zone id
      */
-    public HomeWizardP1MeterHandler(Thing thing) {
+    public HomeWizardP1MeterHandler(Thing thing, ZoneId zoneId) {
```

Comment:
It would be better to store `TimeZoneProvider` here in order to call `timeZoneProvider.getTimeZone()` at the moment when creating `ZonedDateTimes`. This way changes in time-zone will automatically have effect immediately.

Reply:
Yep, totally true.